### PR TITLE
Change : UI Declutter, Show Relevant Data Based on Mission Status

### PR DESCRIPTION
**Pull request by valonm**
**We can merge this if we like it, but for now I just put it together to help drive the UI discussion.**

### Added "Complete" to Mission Status Progression

- Added: `activity.completeTime`

I have re-defined "Complete" from a functional standpoint; there is no change to the underlying data structure. "Complete" now means `completeTime` is set; `endTime` is set now indicates "Closed". 

Differentiating between Complete and Closed provides granularity needed for contextual UI rendering based on Mission Status.

Pending -> Open (for sign in) -> Started -> **Complete** -> Closed

The Mission Status Button on EventPage has been updated to implement revised render sequencing.

- Clicking "Complete" sets the `completeTime`, the button updates to "Close".
- Clicking "Close" sets the `endTime`, the button updates to "Reactivate".
- Clicking "Reactivated" clears `completeTime` and `endTime`, the button updates to "Complete"

### Added Contextual Fields Based on Mission Status Progression

The Mission Details in the top section of the page are dependent on the Mission Status:

- **Pending ("Not Started")**: The Mission Start Time and Standby Responder Count are Shown.
- **Open(for sign in) and In Progress**: The Signed In (enroute) responder Count and Checked In (arrived at base) Responder Count are shown.
- **Complete**: The Demobilized (departed base, turned around) Responder Count and Active (Not Signed Out) Responder Count are Shown.
- **Closed**: No Responder Counts are shown

### Implemented Tabs and Moved Extra Data Fields to the "Detail" tab.

- The "Roster Tab is visible by default.
- The "Data Tab" has full Responder Count Data and other, less important mission details.

### Other UI Tweaks

- Changed the Description to only show the first line, clicking on it expands to toggle the full text.
- Changed the Positioning of the StatusUpdater on larger screens ( moved off to the right, inline with org chips.)
- Removed unnecessary labelling like "Roster:" and "Participating Organizations;"

### Pending Mission
image

### Open (for sign in) Mission
image

### In Progress Mission
image

### Completed Mission
image

### Closed Mission
image

### Mobile
image

### Detail Tab
image

### CODE REVIEW

File: src/app/(main)/EventPage.tsx [hunk 1]
```diff
@@ -68,13 +70,85 @@ const Roster = ({participants, orgs, startTime}: {participants: Record<string, P
...
+      </Dialog>
+    </>
+  );
+
+}
+
+const ActivityStatusSummaryFields = ({activity}:{activity: Activity}) => {
+  const participants = Object.values(activity.participants);
+
+  if (isComplete(activity)) {
```
Comment (davidgut76) [inline src/app/(main)/EventPage.tsx:116]: Why not show all the details when it's closed? Presumably at that point the only reason someone would be looking at a closed mission or event would be to examine all the details.

File: src/components/ActivityTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+const reduceStandby = (count: number, participant: Participant) => {
+  return count + (participant?.timeline[0].status === ResponderStatus.Standby ? 1 : 0);
+}
+
+const reduceSignedIn = (count: number, participant: Participant) => {
+  return count + (participant?.timeline[0].status === ResponderStatus.SignedIn ? 1 : 0);
+}
+
+const reduceCheckedIn = (count: number, participant: Participant) => {
+  return count + (isActive(participant?.timeline[0].status) ? 1 : 0);
```
Comment (davidgut76) [inline src/components/ActivityTabs.tsx:57]: This is the same as reduceActive. Should it be?

File: src/components/ActivityTabs.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
+    setValue(newValue);
+  };
+
+  return (
+    <Box sx={{ width: '100%' }}>
+      
+      <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
+        <Tabs value={value} onChange={handleChange} aria-label="basic tabs example">
```
Comment (davidgut76) [inline src/components/ActivityTabs.tsx:71]: Fix label

File: src/types/participant.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+export function isCheckedIn(status: ResponderStatus) {
+  return [
+    ResponderStatus.Available,
+    ResponderStatus.Assigned
+  ].includes(status);
+}
+
+export const reduceActive = (count: number, participant: Participant) => {
```
Comment (davidgut76) [inline src/types/participant.ts:21]: Many of these functions are also redefined above. Can we have just 1 implementation?

File: src/types/participant.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  return count + (isActive(participant?.timeline[0].status) ? 1 : 0);
+}
+
+export const reduceStandby = (count: number, participant: Participant) => {
+  return count + (participant?.timeline[0].status === ResponderStatus.Standby ? 1 : 0);
```
Comment (davidgut76) [inline src/types/participant.ts:26]: This part is also repeated many times and could go into a helper function: 
count + (participant?.timeline[0].status === [status] ? 1 : 0)

File: src/app/(main)/EventPage.tsx [hunk 6]
```diff
@@ -68,13 +70,85 @@ const Roster = ({participants, orgs, startTime}: {participants: Record<string, P
   )
 }
 
+const activityUpdateOptions = {
+  complete: { label: 'Complete', callback: (activity: Activity) => ActivityActions.complete(activity.id, new Date().getTime()) },
+  close: { label: 'Close', callback: (activity: Activity) => ActivityActions.closed(activity.id, new Date().getTime()) },
```
[the remark this answers is not in the document]
Comment (valonm) [reply src/app/(main)/EventPage.tsx:75]: "close" was telling me it was a keyword, I couldn't use it.

File: src/app/(main)/EventPage.tsx [hunk 7]
```diff
@@ -68,13 +70,85 @@ const Roster = ({participants, orgs, startTime}: {participants: Record<string, P
...
+const activityUpdateOptions = {
+  complete: { label: 'Complete', callback: (activity: Activity) => ActivityActions.complete(activity.id, new Date().getTime()) },
+  close: { label: 'Close', callback: (activity: Activity) => ActivityActions.closed(activity.id, new Date().getTime()) },
+  reactivate: { label: 'Reactivate', callback: (activity: Activity) => ActivityActions.reactivate(activity.id) }
+};
+
+const getActivityUpdateOption = (activity: Activity) => {
+  if (!isActive(activity)) { return activityUpdateOptions.reactivate; }
+  if (isComplete(activity)) { return activityUpdateOptions.close; }
+  return activityUpdateOptions.complete;
```
[the remark this answers is not in the document]
Comment (valonm) [reply src/app/(main)/EventPage.tsx:82]: The other statuses are time based, not manually set.

File: src/components/ActivityTabs.tsx (hunk 2 above)
Comment (valonm) [reply src/components/ActivityTabs.tsx:57]: That should be `isSignedIn`, good catch

File: src/types/participant.ts (hunk 4 above)
Comment (valonm) [reply src/types/participant.ts:21]: Yeah, I'm planning to migrate the references to the other class, and deprecate the definitions in activities. There's a lot though and I my intent here was to get this functional for sharing. If we deploy this ill refactor the other refs.

File: src/types/participant.ts (hunk 5 above)
Comment (valonm) [reply src/types/participant.ts:26]: Same, will refactor duplicated definitions of we're going to merge this.

**Comment by colincwilliams-kcsara** [comment]
Should this be closed at this point? I think with Matt's roster view/tabs update this is outdated?

**Pull request closed without merging**